services/SocialShareService.php

```diff
@@ -0,0 +1,260 @@
+            $enabledPlatforms[] = self::PLATFORM_FACEBOOK;
+        }
+
+        if ($config->get('twitter_enabled', true)) {
+            $enabledPlatforms[] = self::PLATFORM_TWITTER;
+        }
+
+        if ($config->get('linkedin_enabled', true)) {
+            $enabledPlatforms[] = self::PLATFORM_LINKEDIN;
+        }
+
+        if ($config->get('line_enabled', true)) {
+            $enabledPlatforms[] = self::PLATFORM_LINE;
+        }
+
+        if ($config->get('bluesky_enabled', true)) {
+            $enabledPlatforms[] = self::PLATFORM_BLUESKY;
+        }
+
+        return $enabledPlatforms;
+    }
+
+    /**
+     * Generate share link HTML for a specific platform
+     * 
+     * @param string $platform Platform identifier
+     * @param string $permalink Permalink to the content
+     * @param string $description Content description
+     * @param array $options Rendering options
+     * @return string HTML link tag
+     */
+    public function createShareLink($platform, $permalink, $description, $options = [])
+    {
+        if (!isset($this->platforms[$platform])) {
+            throw new \InvalidArgumentException("Unsupported platform: {$platform}");
+        }
+
+        $options = array_merge($this->defaultOptions, $options);
+
+        $url = $this->getSharingUrl($platform, $permalink, $description);
+        $icon = $this->getPlatformIcon($platform);
+
+        return Html::a($icon, $url, $options);
+    }
+
+    /**
+     * Generate HTML for all enabled share links
+     * 
+     * @param mixed $object Content object that provides getContentDescription()
+     * @param string $permalink Permalink to the content
+     * @param array $options Rendering options
+     * @return string HTML with all enabled share links
+     */
+    public function renderShareLinks($object, $permalink, $options = [])
+    {
+        $description = method_exists($object, 'getContentDescription') 
+            ? $object->getContentDescription() 
+            : '';
+
+        $links = [];
+        $enabledPlatforms = $this->getEnabledPlatforms();
+
+        foreach ($enabledPlatforms as $platform) {
+            $links[] = $this->createShareLink($platform, $permalink, $description, $options);
+        }
+
+        return implode("\n", $links);
+    }
+
+    /**
+     * Get HTML for the platform icon
+     * 
+     * @param string $platform Platform identifier
+     * @return string HTML icon
+     */
+    protected function getPlatformIcon($platform)
+    {
+        if (!isset($this->platforms[$platform])) {
+            throw new \InvalidArgumentException("Unsupported platform: {$platform}");
+        }
+
+        $iconClass = $this->platforms[$platform]['iconClass'];
+        $iconColor = $this->platforms[$platform]['iconColor'];
+
+        // Must apply `->right()` else icons are displayed on the left of comments/like WallEntryLinks
+        return Icon::get($iconClass)->color($iconColor)->style('font-size:16px')->right();
```

Comment:
@ArchBlood I don't think we should apply the right alignment to each icon because it sorts them in reverse order.
It is better to add the style class `pull-right` to the wrapper as `return ['class' => 'shareLinkContainer pull-right'];` in the method `ShareLink->getAttributes()`.

PS: Also I don't know why you have there the class `shareLinkContainer` if it doesn't have style definitions, so the method can be changed only to `return ['class' => 'pull-right'];`.

Reply:
> @ArchBlood I don't think we should apply the right alignment to each icon because it sorts them in reverse order. It is better to add the style class `pull-right` to the wrapper as `return ['class' => 'shareLinkContainer pull-right'];` in the method `ShareLink->getAttributes()`.
> 
> PS: Also I don't know why you have there the class `shareLinkContainer` if it doesn't have style definitions, so the method can be changed only to `return ['class' => 'pull-right'];`.

Seems to be working now, so I'm not sure here https://github.com/humhub-contrib/socialshare/pull/12/commits/c17f0ba2ba6448e48f809c9674d6a8490b989d18;
![Screenshot_8](https://github.com/user-attachments/assets/9706283d-0232-46f3-8e95-35ecad97db89)